app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -710,6 +716,11 @@ public BrowserFragment createBrowserFragment(@NonNull String url) {
         return fragment;
     }
 
+    public BrowserFragment createBrowserFragmentForRestoreTab(@NonNull String tabId) {
```

Comment:
In my personal opinion, this is not only for restoring tabs. It is actually for **creating a browser fragment and switch to specific tab**.